web/models/user.ex

```diff
@@ -20,6 +20,7 @@ defmodule Yggdrasil.User do
     model
     |> cast(params, ~w(username password password_confirmation), ~w())
     |> update_change(:username, &String.downcase/1)
+    |> validate_no_spaces(:username)
```

Comment:
I think this could be replaced using the built in [validate_format/4](http://hexdocs.pm/ecto/Ecto.Changeset.html#validate_format/4) function. Something like:

```elixir
|> validate_format(:username, ~r/[^ ], message: "can not contains spaces")
```

You could still wrap it up in a `validate_no_spaces/2` function for clarity though.